library/src/main/java/io/ona/kujaku/views/KujakuMapView.java

```diff
@@ -880,12 +899,60 @@ public void onResume() {
         super.onResume();
         getMapboxMap();
         // This prevents an overlay issue the first time when requesting for permissions
-        if (Permissions.check(getContext(), Manifest.permission.ACCESS_FINE_LOCATION)) {
-            if (getContext() instanceof Activity) {
-                final Activity activity = (Activity) getContext();
-                LocationSettingsHelper.checkLocationEnabled(activity);
-            }
-            warmUpLocationServices();
+        if (warmGps && Permissions.check(getContext(), Manifest.permission.ACCESS_FINE_LOCATION)) {
+            checkLocationSettingsAndStartLocationServices(true);
+        }
+    }
+
+    private void checkLocationSettingsAndStartLocationServices(boolean shouldStartNow) {
+        if (getContext() instanceof Activity) {
+            Activity activity = (Activity) getContext();
+
+            LocationSettingsHelper.checkLocationEnabled(activity, new ResultCallback<LocationSettingsResult>() {
+                @Override
+                public void onResult(LocationSettingsResult result) {
+                    final Status status = result.getStatus();
+                    switch (status.getStatusCode()) {
+                        case LocationSettingsStatusCodes.SUCCESS:
+                            Log.i(TAG, "All location settings are satisfied.");
+
+                            // You can continue warming the GPS
+                            if (shouldStartNow) {
+                                warmUpLocationServices();
+                            }
+                            break;
+                        case LocationSettingsStatusCodes.RESOLUTION_REQUIRED:
+                            Log.i(TAG, "Location settings are not satisfied. Show the user a dialog to upgrade location settings");
+
+                            // This enables us to back-off in case the user has already denied the request
+                            // to turn on location settings
+                            if (!hasAlreadyRequestedEnableLocation) {
+                                hasAlreadyRequestedEnableLocation = true;
+
+                                try {
+                                    // Show the dialog by calling startResolutionForResult(), and check the result
+                                    // in onActivityResult().
+                                    status.startResolutionForResult(activity, Constants.RequestCode.LOCATION_SETTINGS);
+                                } catch (IntentSender.SendIntentException e) {
+                                    Log.i(TAG, "PendingIntent unable to execute request.");
+                                }
+                            } else {
```

Comment:
What does is the intention of this else block?

Could you add comments to better explain the high level objective? It's not that clear.